src/components/Auth/service.js

```diff
@@ -0,0 +1,84 @@
+const jwtFacade = require('./jwt.facade');
+
+const { AuthUserModel, RefreshTokenModel } = require('./models');
+const { getPasswordHash, isPasswordCorrect } = require('./password.helper');
+
+async function getAuthUserById(authUserId) {
+    return AuthUserModel.findById(authUserId).exec();
+}
+
+async function createAuthUser(data) {
+    const passwordHash = await getPasswordHash(data.password);
+    return AuthUserModel.create({
+        ...data,
+        passwordHash,
+    });
+}
+
+async function getAuthUserByEmailPassword(email, password) {
+    const authUser = await AuthUserModel.findOne({ email }).select('+passwordHash').exec();
+    if (
+        authUser === null
+        || await isPasswordCorrect(password, authUser.passwordHash) === false
```

Comment:
no need check equal to boolean if method already return boolean 
thats must work like 
if (authUser === null
        || await isPasswordCorrect(password, authUser.passwordHash))

Reply:
yeah, it looks a bit off, but that's because I wanted to avoid the second condition execution in cases is `authUser` is **null**. 
and `await` inside of the `if` condition reads harder, but there's _false check_, so I can't change it the way you suggested. only -- `!await isPasswordCorrect(password, authUser.passwordHash)` -- and I find it even more confusing like that, so left it as is.